Feature/optional list link

rga-alex-deng opened the pull request:
Attempt to implement feature for an additional relation field

Optional[List[Link[]]]

See issue: https://github.com/roman-right/beanie/issues/300

I was hoping I could get some feedback on this feature if possible @roman-right 

I'm aware that I'm missing some integration testing in the code however we integrated and tested it on a different application and we're not seeing any issues so far.

Thanks for your time!

roman-right commented:
Hey!

Thank you for the PR. The code looks good!

Could you please:
- fix the linter issues. You can check it using pre-commit.
- update the documentation. Here is the place to add the new type: https://github.com/roman-right/beanie/blob/main/docs/tutorial/relations.md

Thank you!

rga-alex-deng commented:
> Hey!
> 
> Thank you for the PR. The code looks good!
> 
> Could you please:
> 
> * fix the linter issues. You can check it using pre-commit.
> * update the documentation. Here is the place to add the new type: https://github.com/roman-right/beanie/blob/main/docs/tutorial/relations.md
> 
> Thank you!

Thank you for the feedback @roman-right !

I've fixed the linter issues and updated the doc. Changes have been pushed to the branch on this PR.
